docs/transformations/DeadCodeRemoval.md

```diff
@@ -0,0 +1,91 @@
+# Dead Code Removal
+
+## Overview
+
+The goal of this transformation is to remove all unnecessary code which is never executed during any call of the function. 
+
+## Example:
+
+```python
+def foo(x):
+    if x >= 0:
+        return x // 10
+        print(x)
+    return -x 
+    print(-x)
+```
+Transforms to:
+```python
+def foo(x):
+    if x >= 0:
+        return x // 10
+    return -x
+```
+
+## Algorithm:
+
+### Using Control Flow Graph
+1. Build [CFG](https://en.wikipedia.org/wiki/Control-flow_graph) for the program.
+2. Traverse the graph in reverse order and search for **unreachable** vertices.
+    The vertex is unreachable if: 
+    * The vertex is not the first in the graph.
+    * There is no incoming edges into this vertex.
+3. Delete the unreachable vertex. Iterate for all outcoming vertices from this vertex and again search for **unreachable** vertices.
```

Comment:
Please, add the exact list of the events that cause unreachable code and check those you support. For example, there are `assert False` or `sys.exit` that you can detect, but you cannot detect this case:
```
a = int(input())
if a == 5:
    print(a)
elif a != 5:
    print(a)
else:
    print(a)
```


It would be also great to refer to some exhaustive and proven list of unreachable code cases.